ee/clickhouse/queries/groups_join_query.py

```diff
@@ -24,16 +22,19 @@ def __init__(
         filter: Union[Filter, PathFilter, RetentionFilter],
         team_id: int,
         column_optimizer: Optional[ColumnOptimizer] = None,
+        join_key: Optional[str] = None,
     ) -> None:
         self._filter = filter
         self._team_id = team_id
         self._column_optimizer = column_optimizer or ColumnOptimizer(self._filter, self._team_id)
+        self._join_key = join_key
 
     def get_join_query(self) -> Tuple[str, Dict]:
         join_queries, params = [], {}
 
         for group_type_index in self._column_optimizer.group_types_to_query:
             var = f"group_index_{group_type_index}"
+            group_join_key = self._join_key or f"$group_{group_type_index}"
```

Comment:
nit: you could also do this in the constructor.

Reply:
how would I get the index? O.o - or do you mean creating a lambda that takes an index?

Reply:
Nevermind :facepalm: